fuse-products/src/main/java/software/tnb/product/csb/application/SpringBootApp.java

```diff
@@ -48,62 +54,86 @@ public boolean shouldRun() {
-
-            LOG.info("Building {} application project", name);
-            Maven.invoke(requestBuilder.build());
+        // If there is an existing jar defined, don't create a new app
+        if (getExistingJar() == null) {
+            if (integrationBuilder instanceof AbstractGitIntegrationBuilder<?>
+                && ((AbstractGitIntegrationBuilder<?>) integrationBuilder).getRepositoryUrl() != null) {
+                mavenGitApp = new MavenGitRepository((AbstractMavenGitIntegrationBuilder<?>) integrationBuilder
+                    , name, getLogPath(Phase.BUILD)
+                    , ((AbstractMavenGitIntegrationBuilder<?>) integrationBuilder).buildProject());
+                shouldRun = ((AbstractGitIntegrationBuilder<?>) integrationBuilder).runApplication();
+            } else {
+                if (integrationBuilder instanceof CamelJBangIntegrationBuilder) {
+                    createUsingJBang();
+                } else {
+                    createUsingMaven();
+                }
+
+                final Path basePath = TestConfiguration.appLocation().resolve(name);
+
+                removeExistingTests(basePath);
+
+                keepWebLayerOnlyForOcp(OpenshiftConfiguration.isOpenshift(), basePath);
+
+                IntegrationGenerator.createFiles(integrationBuilder, basePath);
+
+                customizeMain(basePath);
+
+                customizeDependencies(integrationBuilder.getDependencies());
+
+                createOpenshiftProfileIfNotPresent(basePath);
+
+                customizePlugins(integrationBuilder.getPlugins());
+
+                BuildRequest.Builder requestBuilder = new BuildRequest.Builder()
+                    .withBaseDirectory(basePath)
+                    .withGoals("clean", "package")
+                    .withProperties(Map.of(
+                        "skipTests", "true"
+                    ))
+                    .withLogFile(getLogPath(Phase.BUILD))
+                    .withLogMarker(LogStream.marker(name, Phase.BUILD));
+
+                LOG.info("Building {} application project", name);
+                Maven.invoke(requestBuilder.build());
+            }
         }
     }
 
+    private void createUsingJBang() {
+        List<String> arguments = new ArrayList<>(List.of(
+            "--runtime", "spring-boot",
+            // Align the generated CamelApplication class to correct package
+            "--package-name", TestConfiguration.appGroupId()
+            // todo
+            //                    "--camel-spring-boot-version", SpringBootConfiguration.camelSpringBootVersion(),
+            //                    "--spring-boot-version", SpringBootConfiguration.springBootVersion(),
+        ));
+        super.createUsingJBang(arguments);
+    }
+
+    private void createUsingMaven() {
+        LOG.info("Creating Camel SpringBoot application project for integration {}", name);
+        Map<String, String> properties = new HashMap<>(13);
```

Comment:
why 13? :D I would rather pass the `Map.of(...)` into the constructor :thinking: